react/features/token-data/TokenData.js

```diff
@@ -0,0 +1,92 @@
+/* @flow */
+
+import jwtDecode from 'jwt-decode';
+
+/**
+ * Represent the data parsed from the JWT token
+ */
+export default class TokenData {
+
+    /**
+     * Object representing callee data.
+     */
+    callee: Object;
+
+    /**
+     * Object representing caller data.
+     */
+    caller: Object;
+
+    /**
+     * Contains group.
+     */
+    group: string;
+
+    /**
+     * Indicates whether user is guest.
+     */
+    isGuest: boolean;
+
+    /**
+     * Raw and encoded JWT token.
+     */
+    jwt: string;
+
+    /**
+     * Payload object from parsed JWT token.
+     */
+    payload: Object;
```

Comment:
I removed setting of raw token by default and allowed storing it only when issuer is not anonymous or corresponding token was sent via config parameter. And what about payload: we use payload object outside of the context of TokenData only in one place so we can get rid of storing it as a property too.